autogluon/scheduler/fifo.py

```diff
@@ -105,29 +130,23 @@ class FIFOScheduler(TaskScheduler):
     >>> scheduler.get_training_curves(plot=True)
     """
 
-    def __init__(self, train_fn, args=None, resource=None,
-                 searcher=None, search_options=None,
-                 checkpoint=None,
-                 resume=False, num_trials=None,
-                 time_out=None, max_reward=None, reward_attr='accuracy',
-                 time_attr='epoch',
-                 visualizer='none', dist_ip_addrs=None,
-                 training_history_callback=None,
-                 training_history_callback_delta_secs=60,
-                 delay_get_config=True):
-        super().__init__(dist_ip_addrs)
-        if resource is None:
-            resource = {'num_cpus': 1, 'num_gpus': 0}
-        self.resource = resource
-
-        if searcher is None:
-            searcher = 'random'  # RandomSearcher
+    def __init__(self, train_fn, **kwargs):
```

Comment:
Since this function takes a lot of parameters which have now been moved to kwargs, do we have any intention to raise an exception if an unknown parameter is passed, or will we just ignore them?

Reply:
It only works the (simple) way I did it, if such extra parameters are ignored, because I am just passing all parameters also of the subclass. For example, this is called by HyperbandScheduler.__init__, passing things like max_t, grace_period.

By doing these "imputation" outputs into the log, I feel the user can see what is going on.

Reply:
The only thing I can see which could happen now, is that a user misspells an argument, say passes 'searchr' instead of 'searcher'. In this case, the code would not break, but it would generate an output in the log, saying that 'searcher' is imputed. So, it certainly does not pass silently.

Hmm, in fact maybe I can think of something ...

Reply:
OK, I put in a mechanism to catch invalid parameters.

Reply:
Looks great! I am fine for you to merge.